std/uni.d

```diff
@@ -8843,16 +8850,6 @@ void toLowerInPlace(C)(ref C[] s) @trusted pure
 {
     toCaseInPlace!(LowerTriple)(s);
 }
-// overloads for the most common cases to reduce compile time
-@safe pure /*TODO nothrow*/
```

Comment:
IMHO, if the attribute does not depend on the template arguments, then it should definitely be on the templated function. The only reason to rely on inference is because you have to. Otherwise, you're reducing clarity just to avoid typing.